src/content/en/updates/2018/10/animation-worklet.md

```diff
@@ -0,0 +1,493 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Houdini’s Animation Worklet
+
+{# wf_updated_on: 2018-10-08 #}
+{# wf_published_on: 2018-10-06 #}
+{# wf_tags: houdini,css #}
+{# wf_featured_image: /web/updates/images/2018/10/animation-worklet/social.png #}
+{# wf_featured_snippet: Supercharge your web app’s animations with a new primitive from the Houdini Task Force. #}
+{# wf_blink_components: Blink #}
+
+# Houdini's Animation Worklet {: .page-title }
+## Supercharge your webapp's animations
+
+{% include "web/_shared/contributors/surma.html" %}
+
+**TL;DR:** Animation Worklet allows you to write imperative animations that run
+at the device's native frame rate for that extra buttery jank-free smoothness™,
+make your animations more resilient against main thread jank and can be linked
+to scroll instead of time. Animation Worklet is in Chrome Canary (behind the
+"Experimental Web Platform features" flag) and we are planning an [Origin
+Trial](http://bit.ly/OriginTrialSignup) for Chrome 71. You can start using it as
+a progressive enhancement _today_.
+
+## Another Animation API?
+
+Actually no, it is an extension of what we've already have, and with good
+reason! Bear with me, let's start at the beginning: What do we have? If you want
+to animate any DOM element on the web today, you have 2 ½ choices: [CSS
+Transitions](https://developer.mozilla.org/en-US/docs/Web/CSS/transition) for
+simple A to B transitions, [CSS
+Animations](https://developer.mozilla.org/en-US/docs/Web/CSS/animation) for
+potentially cyclical, more complex time-based animations and [Web Animations
+API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API) (or
+WAAPI) for almost arbitrarily complex animations. [WAAPI's support
+matrix](https://caniuse.com/#feat=web-animation) is looking pretty grim, but
+it's on the way up. Until then, there is a
+[polyfill](https://github.com/web-animations/web-animations-js).
+
+What all these methods have in common is that they are stateless and
+time-driven. But some of the effects developers are trying to do are neither
+time-driven nor stateless. For example the infamous parallax scroller is, as the
+name implies, scroll-driven. [Implementing a performant parallax scroller on the
+web today is surprisingly
+hard.](/web/updates/2016/12/performant-parallaxing)
+
+And what about statelessness? Think about Chrome's address bar on Android, for
+example. If you start scrolling down, it also scrolls out of view. But the
+second you start scrolling up, it comes back into view, even if you are half way
+down that page. The animation depends not only on scroll position, but also on
+your previous scroll direction. It is _stateful_.
+
```

Comment:
Interesting example. I think there are two different things at play in this example:
1. Direction itself is a state: to compute direction one needs to know the *previous* scroll position. 
2. You can think of this animation effect as if it is trying to keep the address bar attached to a particular *baseline position* as page is scrolled. In this model, the *baseline position* is also state. 

Here is how the animation behaves:
  * A) page is scrolling up (swipe down gesture) and address bar is fully shown   => do not move the address bar and instead reset the baseline position.
  * B) page is scrolling down (swipe up gesture) or address bar is not fully shown => move the address bar as if it is attached to the baseline position. 
  * C) scrolling is finished  (i.e., animation)  => reset the baseline to top of the document (so next time if user is scrolling up) it starts coming back into view.

Here both A & C are updating the state but there is a subtle difference: only (A) requires the animation itself to be stateful. (C) can be achieved even if state is kept outside the animation.

I think the example you gave 'But the second you start scrolling up, it comes back into view, even if you are half way down the page' is highlighting (C) but I think the case is more clear if we highlight (A) instead. Honestly this is a subtle distinction and maybe not worth trying to clarify.

Also perhaps worth making it clear that computing direction itself requires state (i.e., previous position). Another example of this is "velocity".

Reply:
You are technically correct. I’d like to lead this with a visual example of statefulness, even if the actual state is something else. When people start building this, they will hopefully discover what to do :D